lib/Notification/Notifier.php

```diff
@@ -119,17 +139,15 @@ public function prepare(INotification $notification, $languageCode): INotificati
 				$room = $this->manager->getRoomById((int) $notification->getObjectId());
 			} catch (RoomNotFoundException $e) {
 				// Room does not exist
-				$this->notificationManager->markProcessed($notification);
-				throw new \InvalidArgumentException('Invalid room');
+				throw new AlreadyProcessedException();
 			}
 		}
 
 		try {
 			$participant = $room->getParticipant($userId);
 		} catch (ParticipantNotFoundException $e) {
 			// Room does not exist
-			$this->notificationManager->markProcessed($notification);
```

Comment:
instead of marking as processed in the prepare method, throw a `AlreadyProcessedException` which will correctly delete the notification on all devices.